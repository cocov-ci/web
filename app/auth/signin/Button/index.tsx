'use client'

import Image from 'next/image'

import { useAuth } from 'app/AuthContext'

import Loading from '../Loading'

import styles from './Button.module.scss'

const SignInButton = () => {
  const { login, loading } = useAuth()

  if (loading) return <Loading />

  return (
    <button className={styles.button} onClick={(): void => login()}>
      <Image
        alt="GitHub logo"
        className={styles.logo}
        height="20"
        src="/icons/github.svg"
        width="20"
      />
      Log in with GitHub
    </button>
  )
}

export default SignInButton
