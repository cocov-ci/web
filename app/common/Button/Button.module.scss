@import "styles/variables";

.button {
  position: relative;
  cursor: pointer;
  color: $colorWhite;
  border: none;
  transition: background-color 0.1s ease-in-out;
  font-variation-settings: "wght" 550;
  border-radius: 23px;

  &.uppercase {
    text-transform: uppercase;
  }

  @mixin button-style($color, $padding: 10px 16px, $fontSize: 14px, $hoverColor: darken($color, 10%)) {
    background-color: $color;
    padding: $padding;
    font-size: $fontSize;

    &:hover {
      background-color: $hoverColor;
    }
  }

  &.primary {
    @include button-style($colorBlue, $padding: 12px 16px, $fontSize: 17px);

    line-height: 21px;
  }

  &.secondary {
    @include button-style($colorGrayDark);
  }

  &.inactive {
    @include button-style(transparent, $hoverColor: darken($colorGrayLighter, 10%));

    color: $colorGrayDark;
  }

  &.danger {
    @include button-style($colorRedMedium);
  }

  &.mini {
    @include button-style($colorGrayDark, $padding: 8px 10px, $fontSize: 12px);
  }
}
