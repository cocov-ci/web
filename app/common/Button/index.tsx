import classNames from 'classnames'
import Link from 'next/link'
import React from 'react'

import Url from 'types/Url'

import styles from './Button.module.scss'

interface ButtonProps {
  children: React.ReactNode
  className?: string
  uppercase?: boolean
  onClick?: () => void
  href?: Url
  style?: 'primary' | 'secondary' | 'inactive' | 'danger' | 'mini'
  disabled?: boolean
}

const Button = ({
  children,
  uppercase,
  className,
  onClick,
  href,
  disabled,
  style = 'primary',
}: ButtonProps) => {
  const baseProps = {
    className: classNames(styles.button, className, styles[style], {
      [styles.uppercase]: uppercase,
    }),
  }

  if (href && !disabled) {
    return (
      <Link href={href}>
        <div {...baseProps}>{children}</div>
      </Link>
    )
  }

  return (
    <button
      {...baseProps}
      disabled={disabled === true}
      onClick={() => onClick && !disabled && onClick()}
    >
      {children}
    </button>
  )
}

export default Button
