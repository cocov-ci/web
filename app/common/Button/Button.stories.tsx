import type { Meta, StoryObj } from '@storybook/react'

import Button from './index'

const meta: Meta<typeof Button> = {
  title: 'Common/Button',
  component: Button,
  tags: ['autodocs'],
}

export default meta

type Story = StoryObj<typeof Button>

export const Primary: Story = {
  args: {
    children: <>Button Label Here</>,
    style: 'primary',
    disabled: false,
  },
}

export const Secondary: Story = {
  args: {
    children: <>Button Label Here</>,
    style: 'secondary',
    disabled: false,
  },
}

export const Inactive: Story = {
  args: {
    children: <>Button Label Here</>,
    style: 'inactive',
    disabled: false,
  },
}

export const Danger: Story = {
  args: {
    children: <>Button Label Here</>,
    style: 'danger',
    disabled: false,
  },
}

export const Mini: Story = {
  args: {
    children: <>Button Label Here</>,
    style: 'mini',
    disabled: false,
  },
}
