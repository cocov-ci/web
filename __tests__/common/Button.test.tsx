import { fireEvent, render, screen } from '@testing-library/react'
import { ReactNode } from 'react'

import Button from 'app/common/Button'

import '@testing-library/jest-dom'

describe('Button', () => {
  const ButtonComponent = ({
    onClick,
    children,
  }: {
    onClick?: () => void
    children: ReactNode
  }) => <Button onClick={onClick}>{children}</Button>

  it('renders Button', () => {
    render(<ButtonComponent>button</ButtonComponent>)

    expect(screen.getByRole('button')).toBeVisible()
  })

  it('triggers onClick event', () => {
    const handleClick = jest.fn()
    render(<ButtonComponent onClick={handleClick}>button</ButtonComponent>)
    fireEvent.click(screen.getByText(/button/i))
    expect(handleClick).toHaveBeenCalledTimes(1)
  })

  it('renders Button snapshots', () => {
    const { container } = render(<ButtonComponent>button</ButtonComponent>)
    expect(container).toMatchSnapshot()
  })
})
