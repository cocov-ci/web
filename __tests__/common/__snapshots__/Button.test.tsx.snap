// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders Button snapshots 1`] = `
<div>
  <button
    class="button"
  >
    button
  </button>
</div>
`;
